ignite/metrics/GAN/fid.py

```diff
@@ -0,0 +1,178 @@
+from typing import Callable, Sequence, Union
+
+import numpy as np
```

Comment:
Numpy and scipy are not dependencies for ignite ? Can we get rid of them ?

Reply:
we need them for matrix square root operation.

Reply:
you can use torch for that

Reply:
Torch doesn't have a function and writing one would be complicated for one. I can write it if you want but it might add a hundred or so lines of extra code.

Reply:
Well, your estimate of 100 lines seems too big.
- https://github.com/pytorch/pytorch/issues/25481
- https://github.com/scipy/scipy/blob/v1.6.3/scipy/linalg/_matfuncs_sqrtm.py#L114-L191

For now, let's keep scipy as dep but in this case we have to make it lazy instead of explicit import:
1) If no scipy => no FID metric import
2) user can import FID and have an issue on init if no scipy